ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/eea/OnChainEeaSendRawTransaction.java

```diff
@@ -0,0 +1,104 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.privacy.methods.eea;
+
+import static org.hyperledger.besu.ethereum.privacy.PrivacyGroupUtil.findOnchainPrivacyGroup;
+
+import org.hyperledger.besu.enclave.types.PrivacyGroup;
+import org.hyperledger.besu.ethereum.api.jsonrpc.RpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.privacy.methods.EnclavePublicKeyProvider;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcError;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import org.hyperledger.besu.ethereum.core.Address;
+import org.hyperledger.besu.ethereum.core.Transaction;
+import org.hyperledger.besu.ethereum.eth.transactions.TransactionPool;
+import org.hyperledger.besu.ethereum.privacy.PrivacyController;
+import org.hyperledger.besu.ethereum.privacy.PrivateTransaction;
+
+import java.util.Optional;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+
+public class OnChainEeaSendRawTransaction extends EeaSendRawTransaction {
+
+  public OnChainEeaSendRawTransaction(
+      final TransactionPool transactionPool,
+      final PrivacyController privacyController,
+      final EnclavePublicKeyProvider enclavePublicKeyProvider) {
+    super(transactionPool, privacyController, enclavePublicKeyProvider);
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.EEA_SEND_RAW_TRANSACTION.getMethodName();
+  }
+
+  @Override
+  Optional<PrivacyGroup> findPrivacyGroup(
+      final PrivacyController privacyController,
+      final Optional<Bytes> maybePrivacyGroupId,
+      final String enclavePublicKey,
+      final PrivateTransaction privateTransaction) {
+    if (maybePrivacyGroupId.isEmpty()) {
+      throw new JsonRpcErrorResponseException(JsonRpcError.ONCHAIN_PRIVACY_GROUP_ID_NOT_AVAILABLE);
+    }
+    final Optional<PrivacyGroup> maybePrivacyGroup =
+        findOnchainPrivacyGroup(
+            privacyController, maybePrivacyGroupId, enclavePublicKey, privateTransaction);
+    if (maybePrivacyGroup.isEmpty()) {
+      throw new JsonRpcErrorResponseException(JsonRpcError.ONCHAIN_PRIVACY_GROUP_DOES_NOT_EXIST);
+    }
+    return maybePrivacyGroup;
+  }
+
+  @Override
+  JsonRpcResponse createPMTAndAddToTxPool(
+      final Object id,
+      final PrivateTransaction privateTransaction,
+      final Optional<PrivacyGroup> maybePrivacyGroup,
+      final Optional<Bytes> maybePrivacyGroupId,
+      final String enclavePublicKey,
+      final Address privacyPrecompiledAddress) {
+    final Bytes privacyGroupId = maybePrivacyGroupId.get();
```

Comment:
We are getting from the optional without checking if the value exists. If by any reason the id doesn't exist, this will cause errors that are hard to debug. It is probably worth adding a check and throwing an exception with an easy to read error msg.

Reply:
It is not checked here, but it is checked in the findPrivacyGroup method ...
I'll put in a comment here that this has already been checked!